frontend/apps/marketing/cicd/3-app/template.yml.erb

```
@@ -848,6 +853,64 @@ Resources:
   # End AWS Cloudfront Resources #
   ################################
 
+  #########################
+  # Start Redis Resources #
+  #########################
+
+  RedisSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Redis access from ECS
+      VpcId: !Ref Vpc
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 6379
+          ToPort: 6379
+          SourceSecurityGroupId: !Ref FargateServiceSecurityGroup
+
+  RedisSubnetGroup:
+    Type: AWS::ElastiCache::SubnetGroup
+    Properties:
+      Description: Marketing Sites Redis Cache Subnet Group
+      SubnetIds:
+<% MarketingSites::Configuration::REGIONS[options[:region].to_sym][:vpc][:private_subnets].each_with_index do |_, index| %>
+        - Ref: VpcPrivateSubnet<%= index %>
+<% end %>
+
+  # This enables keyspace notifications for Redis, which is required for the Next.js caching.
+  # See: https://github.com/trieb-work/nextjs-turbo-redis-cache
+  RedisParameterGroup:
+    Type: AWS::ElastiCache::ParameterGroup
+    Properties:
+      CacheParameterGroupFamily: redis7
+      Description: Enable keyspace notifications
+      Properties:
+        notify-keyspace-events: Exe
+
+  RedisCluster:
+    Type: AWS::ElastiCache::ReplicationGroup
+    Properties:
+      ReplicationGroupDescription: Marketing Sites Redis Cache Replication Group
+      Engine: redis
```

Might be useful to assign a human-readable unique identifier so that anyone browsing Elasticache in the AWS web console can determine what each cluster is being used for. Not sure if this will reliable stay below the 40 character limit, though:

```suggestion
      Engine: redis
      ReplicationGroupId: !Sub "marketing-sites-page-cache-${SubdomainName}"
```

---

Unfortunately `ReplicationGroupId` forbids the use `.` which is in `SubdomainName` and I don't see a good way of replacing the periods in CFN. I think passing in a cleaned string via deploy.rb could work, what do you think?

---

I think `SubdomainName` typically will NOT have a `.` in it. For example, the test CSforALL fully qualified domain name would be `CSforALL.marketingsites.test-code.org` with `SubdomainName` = `CSforALL` and BaseDomain = `marketingsites.test-code.org`.

Using just the Subdomain as a unique identifier has worked well for Javabuilder, under the assumption that there's typically a 1:1 mapping between Base domain names and AWS Accounts.

---

I think this will work but we'll also need to differentiate by environment, so I updated to `ReplicationGroupId: !Sub "marketing-sites-page-cache-${SubdomainName}-${EnvironmentType}"

---

> I think this will work but we'll also need to differentiate by environment, so I updated to
> 
> ```
> ReplicationGroupId: !Sub "marketing-sites-page-cache-${SubdomainName}-${EnvironmentType}"
> ```



> I think this will work but we'll also need to differentiate by environment, so I updated to
> 
> ```
> ReplicationGroupId: !Sub "marketing-sites-page-cache-${SubdomainName}-${EnvironmentType}"
> ```

We currently have a 1:1 mapping between environment type and base domain name and if we firmly adopt a 1:1 mapping between environment type (base domain name) and AWS Account, then we may not need environment type in the identifier which may help in staying below the 40 character limit 😅